lib/Doctrine/DBAL/Event/SchemaColumnDefinitionEventArgs.php

```diff
@@ -33,9 +33,9 @@ class SchemaColumnDefinitionEventArgs extends SchemaEventArgs
     private $connection;
 
     /**
-     * @param mixed[] $tableColumn
-     * @param string  $table
-     * @param string  $database
+     * @param mixed[]     $tableColumn
+     * @param string      $table
+     * @param string|null $database
```

Comment:
Is it possible to not declare it nullable? A column event outside of a database doesn't make much sense.

Reply:
I will give it a try. I didn't change any signatures so far, just documented what it is today as best as I could tell.

Reply:
It is because of this and this https://github.com/doctrine/dbal/blob/develop/lib/Doctrine/DBAL/Driver.php#L53 and https://github.com/doctrine/dbal/blob/develop/lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php#L159

Reply:
Right. Please use something like 933b850a88b1151430046fe56908389cc57f18c1. If there's no DB specified in the connection parameters and not DB is currently selected, the schema manager won't be able to fetch any assets anyways. It should be an error.

Reply:
Can we make that behavior change in another PR?

Reply:
I'd rather make it now in order to not widen the type constraint. I wouldn't object if it already was `?string`. If it's an error reported by PHPStan, it's fine to temporarily suppress the error.